src/components/cl/hier/cl_hier_context.c

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "cl_hier.h"
+#include "utils/ucc_malloc.h"
+
+UCC_CLASS_INIT_FUNC(ucc_cl_hier_context_t,
+                    const ucc_base_context_params_t *params,
+                    const ucc_base_config_t         *config)
+{
+    const ucc_cl_context_config_t *cl_config =
+        ucc_derived_of(config, ucc_cl_context_config_t);
+    ucc_config_names_array_t *tls = &cl_config->cl_lib->tls;
+    ucc_status_t              status;
+    int                       i;
+
+    UCC_CLASS_CALL_SUPER_INIT(ucc_cl_context_t, cl_config->cl_lib,
+                              params->context);
+    if (tls->count == 1 && !strcmp(tls->names[0], "all")) {
```

Comment:
so we have UCC_CL_HIER_TLS, UCC_CL_HIER_NODE_SBGP_TLS and UCC_CL_HIER_NET_SBGP_TLS, what is the relationship between them? if for example HIER_TLS=ucp and NET_SBGP_TLS=sharp, NODE_SBGP_TLS=ucp what TLs will be used?